www/js/diary/list/LabelListScreen.tsx

```diff
@@ -4,53 +4,45 @@ import { Appbar, useTheme } from 'react-native-paper';
 import DateSelect from './DateSelect';
 import FilterSelect from './FilterSelect';
 import TimelineScrollList from './TimelineScrollList';
-import LabelTabContext from '../LabelTabContext';
 import NavBar from '../../components/NavBar';
+import TimelineContext from '../../TimelineContext';
+import { LabelTabContext } from '../LabelTab';
+import { DateTime } from 'luxon';
+import { displayErrorMsg } from '../../plugin/logger';
 
 const LabelListScreen = () => {
-  const {
-    filterInputs,
-    setFilterInputs,
-    timelineMap,
-    displayedEntries,
-    queriedRange,
-    loadSpecificWeek,
-    refresh,
-    pipelineRange,
-    loadAnotherWeek,
-    isLoading,
-  } = useContext(LabelTabContext);
+  const { filterInputs, setFilterInputs, displayedEntries } = useContext(LabelTabContext);
+  const { timelineMap, loadSpecificWeek, timelineIsLoading, refreshTimeline } =
+    useContext(TimelineContext);
   const { colors } = useTheme();
 
   return (
     <>
-      <NavBar>
+      <NavBar isLoading={Boolean(timelineIsLoading)}>
         <FilterSelect
           filters={filterInputs}
           setFilters={setFilterInputs}
           numListDisplayed={displayedEntries?.length}
           numListTotal={timelineMap?.size}
         />
         <DateSelect
-          tsRange={{ oldestTs: queriedRange?.start_ts, latestTs: queriedRange?.end_ts }}
-          loadSpecificWeekFn={loadSpecificWeek}
+          mode="single"
+          onChoose={({ date }) => {
+            const d = DateTime.fromJSDate(date).toISODate();
+            if (!d) return displayErrorMsg('Invalid date');
+            loadSpecificWeek(d);
+          }}
```

Comment:
Yes, that's a good observation. I was wondering that too because the datepickers on both tabs are now controlling the same thing. It would certainly be simpler and more consistent to make both a "range" selection.

My main focus for this PR was refactoring and I didn't want to deviate too far from the current behavior, but we should definitely consider for the future.

My biggest reservation is that a "range" selection on the label tab would add a couple more clicks to the flow.
"Single" date selection takes 1 click: user selects one date and we immediately load the week surrounding that date. This is handy for the label screen to quickly look at some week in the past.
"Range" date selection takes 3 clicks: i) select start date, ii) select end date, iii) click "Save". This gives the user more control over which specific days are going to be shown, which seems valuable for exploring historical metrics data.

Because of this, I think some users might prefer to keep the one-click datepicker on the Label tab.
Although, we might be able to get the "range" selection down to 2 clicks, if we are ok with the datepicker "auto-confirming" when a start and end date have been selected (eliminating the need to click "Save")

@shankari What do you think?

Reply:
@JGreenlee and @jiji14 I think that consistency is better than fewer clicks. Although the 1-click option has fewer clicks, it is not clear to the user what that click does (retrieve data for the week surrounding that date). The range makes it more clear and more flexible - what if I took a 2-week vacation and want to load all the data for that week.

Reply:
Great points. Next time we work on the Label screen we will unify the datepickers TODO